### ISSUE TITLE
Breaking changes in v0.4.0?

### ISSUE THREAD

**Issue by alexlawrence** [deltaskelta/docker-alertd#18]
I downloaded the latest version in order to be able to only check for container existence. However, it seems like alerting does not work at all anymore. Sample config:

```
---
duration: 100				# duration in ms between docker API calls
iterations: 0				# number of iterations to run (0 = run forever)

containers:
  - name: foo
    expectedRunning: true

  - name: bar
    expectedRunning: true

  - name: baz

email:
  active: true
  smtp: some.example.com
  password: ...
  port: 587
  from: alerting@example.com
  subject: "monitoring alert"
  to:
    - xyz@example.com
    - abc@example.com
```

When stopping a container expected to be running, nothing happens. 
Any idea what could be wrong?

**Comment by deltaskelta** [comment]
@alexlawrence `duration` of 100 ms and infinite `iterations` are now default and `0` iterations is a valid value, meaning it loads the config and everything and then never checks the docker daemon therefore no alerts would be triggered. 

You should delete those two lines unless you want to change the default values. 

I decided to change this part of the API because it is still in the 0.x.x phase of versioning, and using pointers to check for existence of this setting was the correct way to do this. Sorry if it caused any inconvenience.

**Comment by alexlawrence** [comment]
Removed both `duration`and `iterations`. Still having the same behavior.

Output:
```
Oct 24 11:25:20 f012 systemd[1]: Started start docker-alertd.
Oct 24 11:25:20 f012 docker-alertd[13281]: Using config file: /home/ansible/.docker-alertd.yaml
Oct 24 11:25:20 f012 docker-alertd[13281]: 2017/10/24 11:25:20 email alerts active
Oct 24 11:25:20 f012 docker-alertd[13281]: 2017/10/24 11:25:20 starting docker-alertd
Oct 24 11:25:20 f012 docker-alertd[13281]: ------------------------------
```

**Comment by deltaskelta** [comment]
you're right, I have reproduced it and I am working on a fix now

**Comment by deltaskelta** [comment]
@alexlawrence the command line flag defaults were making the value 0 by default so I switched it back to the 0 value for `iterations` running the monitor indefinitely. 

It should be fixed now, let me know if you have any problems

**Comment by alexlawrence** [comment]
Thanks for the fast work! Could you please update the binaries?

**Comment by deltaskelta** [comment]
yes, they should be there now

**Comment by alexlawrence** [comment]
Sorry to bother again. Alerting works now but the e-mails look a bit weird:

```
reporting: expected running state: %!t(*bool=0xc4201dd808), current running
state: true: Running check recovered
```

It seems that it does not convert the boolean (pointer?) correctly to a string.

**Comment by deltaskelta** [comment]
right, that should have been caught by my linter, but it was obscured in a function call. Should be fixed now in 0.4.3
